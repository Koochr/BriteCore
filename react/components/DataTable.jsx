import React from "react"
import PropTypes from "prop-types"
import InputForm from "./InputForm.jsx"

export default class DataTable extends React.Component {
  static propTypes = {
  	data: PropTypes.arrayOf(PropTypes.object).isRequired,
  	columns: PropTypes.arrayOf(PropTypes.string).isRequired,
  	editableColumns: PropTypes.arrayOf(PropTypes.string),
  	onModify: PropTypes.func,
  	onHeaderClick: PropTypes.func.isRequired,
		onFocusChange: PropTypes.func.isRequired,
  	sort: PropTypes.object,
		focusedCell: PropTypes.object,
  }

  static defaultProps = {
  	editableColumns: [],
  }

  handleSubmit = (value, rowNumber, column) => {
  	this.props.onModify(value, rowNumber, column)
  }

  renderHeader = columns => columns.map(
  	(column, index) => <th
  		className="table-header-cell"
  		key={index}
  		onClick={() => {this.props.onHeaderClick(column)}}
  	>
  		{column}
  		{this.props.sort?.column === column
      && (this.props.sort.ascending
      	? String.fromCharCode(11015)
      	: String.fromCharCode(11014))}
  	</th>
  )

  renderRow = (row, columns, rowNumber) => {
  	let cells = []
  	for (let column of columns) {
  		cells.push(row[column])
  	}
  	return cells.map((cell, index) =>
  		<td key={index} onClick={() => {this.props.onFocusChange(rowNumber, index)}}>
  			{(
  				this.props.editableColumns.includes(columns[index])
          && this.props.focusedCell?.row === rowNumber
          && this.props.focusedCell?.column === index
  			)
  				? <InputForm
  					initialValue={cell}
  					onSubmit={(value) => {this.handleSubmit(value, rowNumber, columns[index])}}
  					submitButtonName="Save"
  					onCancel={() => {this.props.onFocusChange()}}
  				/>
  				: cell
  			}
  		</td>
  	)
  }

  render () {
  	const { data, columns, } = this.props
  	return <table className="table table-striped table-hover">
  		<thead>
  			<tr>
  				{this.renderHeader(columns)}
  			</tr>
  		</thead>
  		<tbody>
  			{data.map((row, index) => <tr key={index} >{this.renderRow(row, columns, index)}</tr>)}
  		</tbody>
  	</table>
  }
}
