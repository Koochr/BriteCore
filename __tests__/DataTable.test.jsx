/* global it, jest, expect */
import React from "react"
import { shallow, configure, } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import DataTable from "../react/components/DataTable.jsx"

configure({ adapter: new Adapter(), })

const mockColumns = ["id", "name", "description",]
const mockData = [
	{
		id: 1,
		name: "John",
		description: "Is a nice guy",
	},

	{
		id: 44,
		name: "Jack",
		description: "Is a nice guy too",
	},
	{
		id: 77,
		name: "Unknown",
		description: "Suspicious person",
	},
	{
		id: 47,
		name: "Ivan",
		description: "Nothing to say about",
	},
]

it("Should render correctly", () => {
	const component = shallow(
		<DataTable data={mockData} columns={mockColumns} onHeaderClick={jest.fn()} onFocusChange={jest.fn()} />
	)
	expect(component).toMatchSnapshot()
})

it("Should render needed components", () => {
	const component = shallow(
		<DataTable data={mockData} columns={mockColumns} onHeaderClick={jest.fn()} onFocusChange={jest.fn()} />
	)
	const table = component.find("table")
	expect(table).toHaveLength(1)
	const ths = table.find("th")
	expect(ths).toHaveLength(3)
	for (let i=0; i<ths.length; i++) {
		expect(ths.at(i).props().children[0]).toStrictEqual(mockColumns[i])
	}
	const trs = table.find("tr")
	expect(trs).toHaveLength(5)
	for (let i=1; i<trs.length; i++) {
		const tds = trs.at(i).find("td")
		expect(tds).toHaveLength(3)
		for (let j=0; j<tds.length; j++) {
			expect(tds.at(j).props().children).toStrictEqual(mockData[i-1][mockColumns[j]])
		}
	}
})

it("Should handle header clicks", () => {
	const mockHeaderClickHandler = jest.fn()
	const component = shallow(
		<DataTable
			data={mockData}
			columns={mockColumns}
			onHeaderClick={mockHeaderClickHandler}
			onFocusChange={jest.fn()}
		/>
	)
	const ths = component.find("th")
	for (let i=0; i<ths.length; i++) {
		ths.at(i).simulate("click")
		expect(mockHeaderClickHandler).toBeCalledWith(mockColumns[i])
	}
})

it("Should handle focus change", () => {
	const mockFocusHandler = jest.fn()
	const component = shallow(
		<DataTable
			data={mockData}
			columns={mockColumns}
			onHeaderClick={jest.fn()}
			onFocusChange={mockFocusHandler}
		/>
	)
	const trs = component.find("tr")
	for (let i=1; i<trs.length; i++) {
		const tds = trs.at(i).find("td")
		for (let j=0; j<tds.length; j++) {
			tds.at(j).simulate("click")
			expect(mockFocusHandler).toBeCalledWith(i-1, j)
		}
	}
})
