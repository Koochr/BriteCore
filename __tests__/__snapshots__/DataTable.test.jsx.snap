// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataTable
    columns={
      Array [
        "id",
        "name",
        "description",
      ]
    }
    data={
      Array [
        Object {
          "description": "Is a nice guy",
          "id": 1,
          "name": "John",
        },
        Object {
          "description": "Is a nice guy too",
          "id": 44,
          "name": "Jack",
        },
        Object {
          "description": "Suspicious person",
          "id": 77,
          "name": "Unknown",
        },
        Object {
          "description": "Nothing to say about",
          "id": 47,
          "name": "Ivan",
        },
      ]
    }
    editableColumns={Array []}
    onFocusChange={[MockFunction]}
    onHeaderClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              id
            </th>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              name
            </th>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              description
            </th>
          </tr>
        </thead>,
        <tbody>
          <tr>
            <td
              onClick={[Function]}
            >
              1
            </td>
            <td
              onClick={[Function]}
            >
              John
            </td>
            <td
              onClick={[Function]}
            >
              Is a nice guy
            </td>
          </tr>
          <tr>
            <td
              onClick={[Function]}
            >
              44
            </td>
            <td
              onClick={[Function]}
            >
              Jack
            </td>
            <td
              onClick={[Function]}
            >
              Is a nice guy too
            </td>
          </tr>
          <tr>
            <td
              onClick={[Function]}
            >
              77
            </td>
            <td
              onClick={[Function]}
            >
              Unknown
            </td>
            <td
              onClick={[Function]}
            >
              Suspicious person
            </td>
          </tr>
          <tr>
            <td
              onClick={[Function]}
            >
              47
            </td>
            <td
              onClick={[Function]}
            >
              Ivan
            </td>
            <td
              onClick={[Function]}
            >
              Nothing to say about
            </td>
          </tr>
        </tbody>,
      ],
      "className": "table table-striped table-hover",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              id
            </th>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              name
            </th>
            <th
              className="table-header-cell"
              onClick={[Function]}
            >
              description
            </th>
          </tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                id
              </th>,
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                name
              </th>,
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                description
              </th>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "id",
                  false,
                ],
                "className": "table-header-cell",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "id",
                false,
              ],
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "name",
                  false,
                ],
                "className": "table-header-cell",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "name",
                false,
              ],
              "type": "th",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "description",
                  false,
                ],
                "className": "table-header-cell",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "description",
                false,
              ],
              "type": "th",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td
                onClick={[Function]}
              >
                1
              </td>
              <td
                onClick={[Function]}
              >
                John
              </td>
              <td
                onClick={[Function]}
              >
                Is a nice guy
              </td>
            </tr>,
            <tr>
              <td
                onClick={[Function]}
              >
                44
              </td>
              <td
                onClick={[Function]}
              >
                Jack
              </td>
              <td
                onClick={[Function]}
              >
                Is a nice guy too
              </td>
            </tr>,
            <tr>
              <td
                onClick={[Function]}
              >
                77
              </td>
              <td
                onClick={[Function]}
              >
                Unknown
              </td>
              <td
                onClick={[Function]}
              >
                Suspicious person
              </td>
            </tr>,
            <tr>
              <td
                onClick={[Function]}
              >
                47
              </td>
              <td
                onClick={[Function]}
              >
                Ivan
              </td>
              <td
                onClick={[Function]}
              >
                Nothing to say about
              </td>
            </tr>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  onClick={[Function]}
                >
                  1
                </td>,
                <td
                  onClick={[Function]}
                >
                  John
                </td>,
                <td
                  onClick={[Function]}
                >
                  Is a nice guy
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 1,
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "John",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "John",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Is a nice guy",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Is a nice guy",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  onClick={[Function]}
                >
                  44
                </td>,
                <td
                  onClick={[Function]}
                >
                  Jack
                </td>,
                <td
                  onClick={[Function]}
                >
                  Is a nice guy too
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": 44,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 44,
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "Jack",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Jack",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Is a nice guy too",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Is a nice guy too",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  onClick={[Function]}
                >
                  77
                </td>,
                <td
                  onClick={[Function]}
                >
                  Unknown
                </td>,
                <td
                  onClick={[Function]}
                >
                  Suspicious person
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": 77,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 77,
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "Unknown",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Unknown",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Suspicious person",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Suspicious person",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  onClick={[Function]}
                >
                  47
                </td>,
                <td
                  onClick={[Function]}
                >
                  Ivan
                </td>,
                <td
                  onClick={[Function]}
                >
                  Nothing to say about
                </td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": 47,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": 47,
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "Ivan",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Ivan",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": "Nothing to say about",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Nothing to say about",
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                id
              </th>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                name
              </th>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                description
              </th>
            </tr>
          </thead>,
          <tbody>
            <tr>
              <td
                onClick={[Function]}
              >
                1
              </td>
              <td
                onClick={[Function]}
              >
                John
              </td>
              <td
                onClick={[Function]}
              >
                Is a nice guy
              </td>
            </tr>
            <tr>
              <td
                onClick={[Function]}
              >
                44
              </td>
              <td
                onClick={[Function]}
              >
                Jack
              </td>
              <td
                onClick={[Function]}
              >
                Is a nice guy too
              </td>
            </tr>
            <tr>
              <td
                onClick={[Function]}
              >
                77
              </td>
              <td
                onClick={[Function]}
              >
                Unknown
              </td>
              <td
                onClick={[Function]}
              >
                Suspicious person
              </td>
            </tr>
            <tr>
              <td
                onClick={[Function]}
              >
                47
              </td>
              <td
                onClick={[Function]}
              >
                Ivan
              </td>
              <td
                onClick={[Function]}
              >
                Nothing to say about
              </td>
            </tr>
          </tbody>,
        ],
        "className": "table table-striped table-hover",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                id
              </th>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                name
              </th>
              <th
                className="table-header-cell"
                onClick={[Function]}
              >
                description
              </th>
            </tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <th
                  className="table-header-cell"
                  onClick={[Function]}
                >
                  id
                </th>,
                <th
                  className="table-header-cell"
                  onClick={[Function]}
                >
                  name
                </th>,
                <th
                  className="table-header-cell"
                  onClick={[Function]}
                >
                  description
                </th>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "id",
                    false,
                  ],
                  "className": "table-header-cell",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "id",
                  false,
                ],
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "name",
                    false,
                  ],
                  "className": "table-header-cell",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "name",
                  false,
                ],
                "type": "th",
              },
              Object {
                "instance": null,
                "key": "2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "description",
                    false,
                  ],
                  "className": "table-header-cell",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  "description",
                  false,
                ],
                "type": "th",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td
                  onClick={[Function]}
                >
                  1
                </td>
                <td
                  onClick={[Function]}
                >
                  John
                </td>
                <td
                  onClick={[Function]}
                >
                  Is a nice guy
                </td>
              </tr>,
              <tr>
                <td
                  onClick={[Function]}
                >
                  44
                </td>
                <td
                  onClick={[Function]}
                >
                  Jack
                </td>
                <td
                  onClick={[Function]}
                >
                  Is a nice guy too
                </td>
              </tr>,
              <tr>
                <td
                  onClick={[Function]}
                >
                  77
                </td>
                <td
                  onClick={[Function]}
                >
                  Unknown
                </td>
                <td
                  onClick={[Function]}
                >
                  Suspicious person
                </td>
              </tr>,
              <tr>
                <td
                  onClick={[Function]}
                >
                  47
                </td>
                <td
                  onClick={[Function]}
                >
                  Ivan
                </td>
                <td
                  onClick={[Function]}
                >
                  Nothing to say about
                </td>
              </tr>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td
                    onClick={[Function]}
                  >
                    1
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    John
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Is a nice guy
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "John",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "John",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Is a nice guy",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Is a nice guy",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td
                    onClick={[Function]}
                  >
                    44
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Jack
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Is a nice guy too
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": 44,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 44,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Jack",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Jack",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Is a nice guy too",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Is a nice guy too",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td
                    onClick={[Function]}
                  >
                    77
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Unknown
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Suspicious person
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": 77,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 77,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Unknown",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Unknown",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Suspicious person",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Suspicious person",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td
                    onClick={[Function]}
                  >
                    47
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Ivan
                  </td>,
                  <td
                    onClick={[Function]}
                  >
                    Nothing to say about
                  </td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "host",
                  "props": Object {
                    "children": 47,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": 47,
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ivan",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Ivan",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": "2",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nothing to say about",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Nothing to say about",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
